hardware/Netatmo.cpp

```diff
@@ -504,8 +674,167 @@ uint64_t CNetatmo::UpdateValueInt(int HardwareID, const char* ID, unsigned char
                 Log(LOG_NORM, szLogString);
         }
         m_mainworker.sOnDeviceReceived(m_HwdID, DeviceRowIdx, devname, nullptr);
-        m_notifications.CheckAndHandleNotification(DeviceRowIdx, m_HwdID, ID, devname, unit, devType, subType, nValue, sValue);
+        m_notifications.CheckAndHandleNotification(DeviceRowIdx, m_HwdID, sDeviceID.c_str(), devname, unit, devType, subType, nValue, sValue);
```

Comment:
no need for the c_str() here

Reply:
This is probably Jurgen's part of the code, but I'll take a look.

Reply:
In the end, it was me: I've changed this because of the upper case conversion needed to compare the device IDs. Indeed, the c_str() function is not needed, as the parameter for the called function is also std::string.

But now I realize that this converted uppercase HEX value is also stored in the database, and this was not the intention of the author (@JurgenLB). So when he wants this to be corrected, I need to see if this has any consequences on the matching process: Original device IDs were in the format "04X" (4 uppercase HEX digits).

Reply:
I moved the uppercase translation, so the updates are done without it, but found that @JurgenLB already changed this in his part of the code. So, in fact, this change has no effect on the results: I'm happy with it.